backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/search/ScrollIterable.java

```diff
@@ -20,56 +20,67 @@
 package org.apache.james.backends.es.search;
 
 import java.util.Iterator;
+import java.util.concurrent.CompletableFuture;
 import java.util.stream.Stream;
 
+import org.apache.james.backends.es.ListenerToFuture;
 import org.apache.james.util.streams.Iterators;
-import org.elasticsearch.action.ListenableActionFuture;
-import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.elasticsearch.action.search.SearchRequest;
 import org.elasticsearch.action.search.SearchResponse;
-import org.elasticsearch.client.Client;
+import org.elasticsearch.action.search.SearchScrollRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.elasticsearch.common.unit.TimeValue;
 
 public class ScrollIterable implements Iterable<SearchResponse> {
+    private static final TimeValue TIMEOUT = TimeValue.timeValueMinutes(1);
 
-    private static final TimeValue TIMEOUT = new TimeValue(60000);
-    private final Client client;
-    private final SearchRequestBuilder searchRequestBuilder;
+    private final RestHighLevelClient client;
+    private final SearchRequest searchRequest;
 
-    public ScrollIterable(Client client, SearchRequestBuilder searchRequestBuilder) {
+    public ScrollIterable(RestHighLevelClient client, SearchRequest searchRequest) {
         this.client = client;
-        this.searchRequestBuilder = searchRequestBuilder;
+        this.searchRequest = searchRequest;
     }
 
     @Override
     public Iterator<SearchResponse> iterator() {
-        return new ScrollIterator(client, searchRequestBuilder);
+        return new ScrollIterator(client, searchRequest);
     }
 
     public Stream<SearchResponse> stream() {
         return Iterators.toStream(iterator());
     }
 
     public static class ScrollIterator implements Iterator<SearchResponse> {
+        private final RestHighLevelClient client;
+        private CompletableFuture<SearchResponse> searchResponseFuture;
```

Comment:
Why not using Reactor here?

Reply:
I think using Mono here is verbose, the CompletableFuture is returned by ElasticSearch client response (so anyway we cannot avoid that), and the future's features is enough for the implementation.